c10/core/impl/CUDATrace.h

```diff
@@ -0,0 +1,20 @@
+#pragma once
+
+#include <c10/core/impl/PyInterpreter.h>
+
+namespace c10 {
+namespace impl {
+
+struct C10_API CUDATrace {
+  // When PyTorch migrates to C++20+, this should be changed to an atomic flag.
+  static std::atomic<const PyInterpreter*> cudaTraceState;
+
+  static void set_trace(const PyInterpreter*);
```

Comment:
It would be good to document that this only works once

Reply:
Done!